### PR TITLE
improvements of the PersistenceTestKit doc samples

### PR DESCRIPTION
**Pull request by patriknw**
* and actually run the tests

### CODE REVIEW

**Review by johanandren — APPROVED** [review]

File: akka-docs/src/test/scala/docs/persistence/testkit/TestKitExamples.scala [hunk 1]
```diff
@@ -4,104 +4,116 @@
...
-        }
-      } else {
-        ProcessingSuccess
+}
+//#test
+
+//#set-event-storage-policy
+import akka.persistence.testkit._
+
+class SampleEventStoragePolicy extends EventStorage.JournalPolicies.PolicyType {
```
Comment (renatocaval) [inline akka-docs/src/test/scala/docs/persistence/testkit/TestKitExamples.scala:69]: I didn't see any example demoing the policies. Is that something to add to the `persistentTestKit`. Is `withPolicy()` a method on `PersistenceTestKit`? I guess so, but not so sure that the documentation is making that clear for the users. Maybe something to improve on another PR.

**Review by octonato — APPROVED** [review]
LGTM, left a small comment though.

File: akka-docs/src/test/scala/docs/persistence/testkit/TestKitExamples.scala (hunk 1 above)
Comment (patriknw) [reply akka-docs/src/test/scala/docs/persistence/testkit/TestKitExamples.scala:69]: @manonthegithub is this something you could follow up on?

Comment (manonthegithub) [reply akka-docs/src/test/scala/docs/persistence/testkit/TestKitExamples.scala:69]: @renatocaval @patriknw it is briefly described in the docs (that you can implement own policy and set it with `withPolicy` method), but there is no example. It seemed straightforward to me. Should we add an example too?

Comment (manonthegithub) [reply akka-docs/src/test/scala/docs/persistence/testkit/TestKitExamples.scala:69]: If yes, should it be separate branch + PR, or you think of it as part of this PR? If separate, then I would better wait for this one to be merged.

Comment (manonthegithub) [reply akka-docs/src/test/scala/docs/persistence/testkit/TestKitExamples.scala:69]: @renatocaval @patriknw I meant that there are no examples of how to set policies, but there are of course how to implement.

Comment (manonthegithub) [reply akka-docs/src/test/scala/docs/persistence/testkit/TestKitExamples.scala:69]: no prob

Comment (manonthegithub) [reply akka-docs/src/test/scala/docs/persistence/testkit/TestKitExamples.scala:69]: did you create an issue?

Comment (manonthegithub) [reply akka-docs/src/test/scala/docs/persistence/testkit/TestKitExamples.scala:69]: added #28869

**Pull request merged**